.github/workflows/e2e_tests.yml

```diff
@@ -17,6 +17,10 @@ jobs:
   e2e:
     runs-on: ubuntu-latest
 
+    permissions:
+      contents: 'read'
+      id-token: 'write'
```

Comment:
Why do we need `id-token: 'write'` permission?

Reply:
this is the default permissions required by upload-cloud-storage action as shown in [this example](https://github.com/google-github-actions/upload-cloud-storage?tab=readme-ov-file#authorization)
searching further, [this document](https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/about-security-hardening-with-openid-connect#adding-permissions-settings) details that this permission is required in order to perform JWT authentication